src/App.js

```diff
@@ -1,25 +1,72 @@
 import logo from './logo.svg';
 import './App.css';
+import React, { useState } from 'react';
+import { getValue } from '@testing-library/user-event/dist/utils';
+import { useRef } from "react";
+export default App;
 
 function App() {
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [gender, setGender] = useState('');
+
+
+  const handleNameChange = (e) => {
+    setName(e.target.value);
+  }
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  }
+
+  const handleGenderChange = (e) => {
+    setGender(e.target.value);
+  }
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log('Name:', name);
+    console.log('Email:', email);
+    console.log('Gender:', gender);
+  }
+
+
+
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+    <React.Fragment>
 
-export default App;
+      <form onSubmit={handleSubmit}>
+
+        <input type="text" name="name" placeholder='Your name' value={name} onChange={handleNameChange} />
+        <br />
+        <input type="email" name="email" placeholder="Your email" value={email} onChange={handleEmailChange} />
+        <br />
+        <input type="radio" value="male" checked={gender === 'male'} onChange={handleGenderChange} />Male
+        <input type="radio" value="female" checked={gender === 'female'} onChange={handleGenderChange} />Female
+        <br />
+        <button type="submit">Submit</button>
```

Comment:
use this button for submitting values. change type and pass callback function here. 

also, as a bonus. validate if all data in the form was provided. if not then disable the button: disabled={true}. 
otherwise button should be available. 

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled
https://react.dev/reference/react-dom/components/input#props -> disabled